lib/shipit/command.rb

```diff
@@ -215,5 +214,40 @@ def parse_arguments(arguments)
       end
       return args, options
     end
+
+    def running?
+      !!pid && !@status
+    end
+
+    def code
+      @status&.exitstatus
+    end
+
+    def signaled?
+      @status.signaled?
+    end
+
+    def reap_child!(block: true)
+      return @status if @status
+      return unless running? # Command was never started e.g. permission denied, not found etc
+      if block
+        _, @status = Process.waitpid2(@pid)
+      elsif res = Process.waitpid2(@pid, Process::WNOHANG)
```

Comment:
Does this return nil/false when the process is still running?

Reply:
Yes it returns `nil`.